### ISSUE TITLE
Null pointer to constant cstring, should be compiler error?

### ISSUE THREAD

**Issue by jockus** [odin-lang/Odin#344]
```
package main
import "core:fmt"
main :: proc(){
        fmt.println(cast(^u8) cstring("test"));
        not_constant := cstring("test");
        fmt.println(cast(^u8) not_constant);
}
```

Outputs:

0x0
0x7FF705DCF88D

Is this the expected behaviour, or should the compiler be erroring on taking the address of the constant cstring?

**Comment by gingerBill** [comment]
This is not expected behaviour.

**Comment by gingerBill** [comment]
Fixed in 0e1cfa5

It will require a better error message later
